streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorTopologyTest.java

```diff
@@ -247,6 +247,17 @@ public void testDrivingInternalRepartitioningTopology() {
     }
 
     @Test
+    public void testDrivingInternalRepartitioningTimestampTopology() {
+        driver = new ProcessorTopologyTestDriver(config, createInternalRepartitioningWithValueTimestampTopology());
+        driver.process(INPUT_TOPIC_1, "key1", "value1@1000", STRING_SERIALIZER, STRING_SERIALIZER);
+        driver.process(INPUT_TOPIC_1, "key2", "value2@2000", STRING_SERIALIZER, STRING_SERIALIZER);
+        driver.process(INPUT_TOPIC_1, "key3", "value3@3000", STRING_SERIALIZER, STRING_SERIALIZER);
+        assertNextOutputRecordTimestamp(OUTPUT_TOPIC_1, "key1", "value1", 1000L);
```

Comment:
Looks like these could be replaced with:
`assertThat(driver.readOutput(...), equalTo(new ProducerRecord<>(topic, null, timestamp, key, value)`